docs/adr/0012_construct_as_transformation.md

```diff
@@ -0,0 +1,441 @@
+  registry based on the *capability* and then select the actual plugin based 
+  on the *type*.
+- It is harder to introduce arbitrary plugins as a plugin registry for the
+  corresponding *capability* has to be available.
+
+**Conclusion** 
+- Generally, this additional layer of indirection adds 
+flexibility at the cost of complexity. This is especially visible in simple use 
+cases (capability with a single type). 
+- However, since we already have this 
+plugin system in place, the complexity of maintaining two different plugin 
+systems is even higher.
+
+#### Resource as Atomic Unit in OCM
+- The atomic unit in ocm is **resource** NOT A PLAIN BLOB or ACCESS, kind of 
+  like the atomic unit in kubernetes are pods not containers.
+- While this takes the operations to a higher abstraction level, offering each
+  operation the ability to add or modify the metadata of a resource.
+
+> [!NOTE]
+> In ocm v1, several interfaces were built against plain blobs or access as 
+> the atomic unit. This led to issues, as 
+> - several operations had to be performed twice (digest calculation of a blob) 
+>   or as the metadata could not be passed along.
+> - several extension points were not flexible enough (resource uploader could
+>   not provide the digest of the uploaded blob, as it only returned a blob)
+
+#### Static Typing
+- The example above shows the *outputs* of each transformation as comments. This
+  is purely for illustration purposes. In the actual implementation, every
+  capability has to define a static output schema (kind of like kubernetes 
+  resource status).
+- The transformation specification is statically typed. This means that
+  everything that we programmatically write into MUST be typed
+  (e.g. `repository: ${attributes.repositorySpec}`, here we have to be able to
+  match the type of `repositorySpec` with the expected type of `repository`).
+- This is required to enable static type analysis, which will be important to
+  avoid problems in complex graphs.
+
+## Comparison with Transformation ADR
+- Instead of expecting a component descriptor as a starting point for all 
+  operations that is consecutively modified and then merged (to detect 
+  conflicts), we create a new component descriptor to be uploaded 
+  with the resources created in the previous operations.
+- This simplifies the operations, as we do not have to deal with 
+  merging and conflict detection.
+- This would also be adopted for the transfer use case. Here, the component 
+  to be uploaded at the target location would be created from scratch and filled
+  with information from the original component descriptor through `cel` expressions.
+
+## Implementation
+
+### Input to Access Types
+
+As mentioned above, there will no longer be an implementation for each input 
+type. Instead, we want new `access types` (e.g. `file`, `dir`, ...) and 
+`uploaders`.
+
+### Plugin Registries
+
+To keep the complexity in check, we assume the transformation implementation 
```

Comment:
I think you should write a concrete example of the mapping to make sure people understand and how this can work